multichain-testing/scripts/fast-usdc.ts

```diff
@@ -0,0 +1,224 @@
+const agoricNamesQ = (vsc: VStorageClient) =>
+  harden({
+    brands: <K extends AssetKind>(_assetKind: K) =>
+      vsc
+        .queryData('published.agoricNames.brand')
+        .then(pairs => Object.fromEntries(pairs) as Record<string, Brand<K>>),
+  });
+
+const fastLPQ = (vsc: VStorageClient) =>
+  harden({
+    metrics: () =>
+      vsc.queryData(`published.fastUsdc.poolMetrics`) as Promise<PoolMetrics>,
+    info: () =>
+      vsc.queryData(`published.${contractName}`) as Promise<{
+        poolAccount: string;
+        settlementAccount: string;
+      }>,
+  });
+
+const parseCommandLine = () => {
+  const { values, positionals } = parseArgs({
+    options: {
+      oracle: {
+        type: 'string',
+      },
+      mnemonic: {
+        type: 'string',
+      },
+      help: {
+        type: 'boolean',
+      },
+    },
+    allowPositionals: true,
+  });
+
+  if (values.help) {
+    console.log(`
+Usage:
+  No arguments      - start the contract and fund the liquidity pool
+  start             - only start the contract
+  fund-pool         - fund the FUSDC Liquidity Pool
+  provision-wallet  - provision a smart wallet (requires --mnemonic)
+  fund-faucet       - fund the faucet account with bridged USDC
+  --oracle          - Comma-separated list of oracle addresses
+
+Examples:
+  ./fast-usdc.ts
+  ./fast-usdc.ts --oracle oracle1:addr1,oracle2:addr2
+  ./fast-usdc.ts start
+  ./fast-usdc.ts start --oracle oracle1:addr1,oracle2:addr2
+  ./fast-usdc.ts fund-pool
+  ./fast-usdc.ts fund-faucet
+`);
+    process.exit(0);
+  }
+
+  const command = positionals[0];
+  const mnemonic = values.mnemonic;
+  const suppliedOracles = values.oracle?.split(',');
+  const oracles = suppliedOracles || [
+    'oracle1:agoric1yupasge4528pgkszg9v328x4faxtkldsnygwjl',
+    'oracle2:agoric1dh04lnl7epr7l4cpvqqprxvam7ewdswj7yv6ep',
+    'oracle3:agoric1ujmk0492mauq2f2vrcn7ylq3w3x55k0ap9mt2p',
+  ];
+
+  return { command, mnemonic, oracles, provisionOracles: true };
+};
+
+const main = async () => {
+  const { command, mnemonic, oracles, provisionOracles } = parseCommandLine();
+  const {
+    chainInfo,
+    commonBuilderOpts,
+    deleteTestKeys,
+    faucetTools,
+    provisionSmartWallet,
+    setupTestKeys,
+    startContract,
+    vstorageClient,
+  } = await commonSetup(mockT, '../config.fusdc.yaml');
+
+  const assertProvisioned = async address => {
+    try {
+      await vstorageClient.queryData(`published.wallet.${address}.current`);
+      return true;
+    } catch {
+      throw new Error(`${address} is not provisioned`);
+    }
+  };
+
+  const provisionWallet = async (mnemonic: string) => {
+    // provision-one must be called by the owner, so we need to add the key to the test keyring
+    const keyname = 'temp';
+    const address = (await setupTestKeys([keyname], [mnemonic]))[keyname];
+    await provisionSmartWallet(address, {
+      BLD: 100n,
+      IST: 100n,
+    });
+    await deleteTestKeys([keyname]);
```

Comment:
seems like there should be a try/finally to prevent `temp` from getting stuck in the test keyring